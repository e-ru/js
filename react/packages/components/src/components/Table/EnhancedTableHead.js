import React from "react";
import PropTypes from "prop-types";

import TableHead from "@material-ui/core/TableHead";
import TableRow from "@material-ui/core/TableRow";
import TableCell from "@material-ui/core/TableCell";
import Checkbox from "@material-ui/core/Checkbox";
import TableSortLabel from "@material-ui/core/TableSortLabel";

const EnhancedTableHead = props => {
  const { onSelectAllClick, order, orderBy, numSelected, rowCount, onRequestSort, headRows } = props;
  const createSortHandler = property => event => {
    onRequestSort(event, property);
  };

  // TODO: <TableCell padding="checkbox" style={{ minWidth: "125px", maxWidth: "125px" }}>
  // width range is needed because next to the checkbox is a edit icon located
  // some kind of a hack...

  return (
    <TableHead>
      <TableRow>
        <TableCell padding="checkbox" style={{ minWidth: "125px", maxWidth: "125px" }}>
          <Checkbox
            indeterminate={numSelected > 0 && numSelected < rowCount}
            checked={numSelected === rowCount}
            onChange={onSelectAllClick}
            inputProps={{ "aria-label": "Select all desserts" }}
          />
        </TableCell>
        {headRows.map(row => (
          <TableCell
            key={row.id}
            align={row.numeric ? "right" : "left"}
            padding={row.disablePadding ? "none" : "default"}
            sortDirection={orderBy === row.id ? order : false}
          >
            <TableSortLabel active={orderBy === row.id} direction={order} onClick={createSortHandler(row.id)}>
              {row.label}
            </TableSortLabel>
          </TableCell>
        ))}
      </TableRow>
    </TableHead>
  );
};

EnhancedTableHead.propTypes = {
  numSelected: PropTypes.number.isRequired,
  order: PropTypes.string.isRequired,
  orderBy: PropTypes.string.isRequired,
  rowCount: PropTypes.number.isRequired,
  headRows: PropTypes.array.isRequired,
  onRequestSort: PropTypes.func.isRequired,
  onSelectAllClick: PropTypes.func.isRequired,
};

export default EnhancedTableHead;
