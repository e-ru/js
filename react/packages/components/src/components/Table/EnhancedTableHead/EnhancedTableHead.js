import React from "react";
import PropTypes from "prop-types";

import { makeStyles } from "@material-ui/core/styles";
import TableHead from "@material-ui/core/TableHead";
import TableRow from "@material-ui/core/TableRow";
import TableCell from "@material-ui/core/TableCell";
import Checkbox from "@material-ui/core/Checkbox";
import TableSortLabel from "@material-ui/core/TableSortLabel";

const useStyles = makeStyles(() => ({
  sticky: {
    position: "sticky",
    top: "0px",
    background: "#fff",
    zIndex: "10",
  },
}));

const EnhancedTableHead = props => {
  const { onSelectAllClick, order, orderBy, numSelected, rowCount, onRequestSort, headRows } = props;
  const createSortHandler = property => event => {
    onRequestSort(event, property);
  };

  // TODO: <TableCell padding="checkbox" style={{ minWidth: "125px", maxWidth: "125px" }}>
  // width range is needed because next to the checkbox is a edit icon located
  // some kind of a hack...

  const classes = useStyles();
  return (
    <TableHead>
      <TableRow>
        <TableCell className={classes.sticky} padding="checkbox" style={{ minWidth: "125px", maxWidth: "125px" }}>
          <Checkbox
            indeterminate={numSelected > 0 && numSelected < rowCount}
            checked={numSelected === rowCount}
            onChange={onSelectAllClick}
            inputProps={{ "aria-label": "Select all desserts" }}
          />
        </TableCell>
        {headRows.map(row => (
          <TableCell
            className={classes.sticky}
            key={row.id}
            align={row.numeric ? "right" : "left"}
            padding={row.disablePadding ? "none" : "default"}
            sortDirection={orderBy === row.id ? order : false}
          >
            <TableSortLabel active={orderBy === row.id} direction={order} onClick={createSortHandler(row.id)}>
              {row.label}
            </TableSortLabel>
          </TableCell>
        ))}
      </TableRow>
      <TableRow style={{ height: "1px", borderBottom: "1px solid rgba(224, 224, 224, 1)" }} />
    </TableHead>
  );
};

EnhancedTableHead.propTypes = {
  numSelected: PropTypes.number.isRequired,
  order: PropTypes.string.isRequired,
  orderBy: PropTypes.string.isRequired,
  rowCount: PropTypes.number.isRequired,
  headRows: PropTypes.array.isRequired,
  onRequestSort: PropTypes.func.isRequired,
  onSelectAllClick: PropTypes.func.isRequired,
};

export default EnhancedTableHead;
