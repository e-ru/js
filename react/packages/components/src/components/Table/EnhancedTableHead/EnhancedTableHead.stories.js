import React from "react";
import { storiesOf } from "@storybook/react";

import EnhancedTableHead from "./EnhancedTableHead";

storiesOf("EnhancedTableHead", module).add("main item", () => (
  <EnhancedTableHead
    numSelected={0}
    order="asc"
    orderBy="username"
    rowCount={5}
    headRows={[
      { id: "row1", numeric: false, disablePadding: true, label: "row1" },
      { id: "row2", numeric: false, disablePadding: false, label: "row2" },
    ]}
    onRequestSort={() => console.log("onRequestSort clicked")}
    onSelectAllClick={() => console.log("onSelectAllClick clicked")}
  />
));
