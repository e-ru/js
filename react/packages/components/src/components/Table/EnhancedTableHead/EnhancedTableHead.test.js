import React from "react";
import { configure, shallow } from "enzyme";
import Adapter from "enzyme-adapter-react-16";

import EnhancedTableHead from "./EnhancedTableHead";

configure({ adapter: new Adapter() });

function setup() {
  const props = {
    numSelected: 0,
    order: "asc",
    orderBy: "username",
    rowCount: 5,
    headRows: [
      { id: "row1", numeric: false, disablePadding: true, label: "row1" },
      { id: "row2", numeric: false, disablePadding: false, label: "row2" },
    ],
    onRequestSort: jest.fn(),
    onSelectAllClick: jest.fn(),
  };

  const wrapper = shallow(<EnhancedTableHead {...props} />);

  return {
    wrapper,
  };
}

describe("<EnhancedTableHead />", () => {
  test("renders", () => {
    const { wrapper } = setup();
    expect(wrapper).toMatchSnapshot();
  });
});
