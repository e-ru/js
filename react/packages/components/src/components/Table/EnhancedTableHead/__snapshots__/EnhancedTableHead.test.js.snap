// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<EnhancedTableHead /> renders 1`] = `ShallowWrapper {}`;
